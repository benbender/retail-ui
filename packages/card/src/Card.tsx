import clsx from 'clsx'
import * as React from 'react'

import { CardColor, CardProvider } from './CardContext'
import { CardStyles } from './styles'

type ReactDivProps = React.HTMLAttributes<HTMLDivElement>
type Ref = HTMLDivElement

export type CardProps = {
  color?: CardColor
  isLoading?: boolean
  isStatic?: boolean
}

export const Card = React.forwardRef<Ref, ReactDivProps & CardProps>(
  (props, ref) => {
    const {
      children,
      className,
      color = 'default',
      isLoading = false,
      isStatic = false,
      ...rest
    } = props

    const cls = clsx(
      className,
      CardStyles.base,
      !isStatic && CardStyles.hov,
      CardStyles.color[color],
    )

    const cardValue = React.useMemo(() => {
      return { color, isLoading }
    }, [color, isLoading])

    return (
      <CardProvider value={cardValue}>
        <div ref={ref} className={cls} {...rest}>
          {children}
        </div>
      </CardProvider>
    )
  },
)

Card.displayName = 'Card'
