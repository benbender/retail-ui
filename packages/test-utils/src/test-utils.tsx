import { defaultTheme, ThemeProvider } from '@retail-ui/theme'
import {
  render as RtlRender,
  RenderOptions,
  RenderResult,
} from '@testing-library/react'
import * as React from 'react'

type UI = Parameters<typeof RtlRender>[0]

// UI-less passthrough fallback to prevent using conditional logic in render
function ChildrenPassthrough({ children }: { children: React.ReactElement }) {
  return children
}

export interface TestOptions extends Omit<RenderOptions, 'wrapper'> {
  /**
   * optional additional wrapper, e.g. Context
   *
   * @example
   * ```ts
   * // single wrapper
   * render(<MyConponent />, {
   *  wrapper: MyContext
   * });
   *
   * // multiple wrapper
   * render(<MyConponent />, {
   *  wrapper: ({ children }) => (
   *    <ContextA>
   *      <ContextB>
   *        {children}
   *      <ContextB />
   *    <ContextA />
   *  )
   * });
   *
   * ```
   */
  wrapper?: typeof ChildrenPassthrough
}

/**
 * Custom render for @testing-library/react
 *
 * @see https://testing-library.com/docs/react-testing-library/setup#custom-render
 * @param component the component under test
 * @param options customized test options
 */
export const render = (
  ui: UI,
  { wrapper: Wrapper = ChildrenPassthrough, ...options }: TestOptions = {},
): RenderResult => {
  return RtlRender(
    <ThemeProvider value={{ theme: defaultTheme }}>
      <Wrapper>{ui}</Wrapper>
    </ThemeProvider>,
    options,
  )
}

export { fireEvent } from '@testing-library/react'
