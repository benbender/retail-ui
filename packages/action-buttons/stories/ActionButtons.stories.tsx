import * as React from 'react'

import { ActionButtons, ActionButtonsProps } from '../src'

const meta = {
  title: 'Retail-UI/ActionButtons',
  component: ActionButtons,
}

export default meta

const Template = (args: ActionButtonsProps) => (
  <ActionButtons {...args} data-testid={meta.title} />
)

export const Simple = Template.bind({})
// @ts-ignore
Simple.args = {}
