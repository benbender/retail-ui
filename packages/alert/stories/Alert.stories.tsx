import * as React from 'react'

import { Alert, AlertProps } from '../src'

const meta = {
  title: 'Retail-UI/Alert',
  component: Alert,
}

export default meta

const Template = (args: AlertProps) => (
  <Alert {...args} data-testid={meta.title} />
)

export const Simple = Template.bind({})
// @ts-ignore
Simple.args = {
  children: 'A alert that informs you of stuff',
}
