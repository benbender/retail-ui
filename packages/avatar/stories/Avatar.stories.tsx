import * as React from 'react'

import { Avatar, AvatarProps } from '../src'

const meta = {
  title: 'Retail-UI/Avatar',
  component: Avatar,
}

export default meta

const Template = (args: AvatarProps) => (
  <Avatar {...args} data-testid={meta.title} />
)

export const Simple = Template.bind({})
// @ts-ignore
Simple.args = {
  src: `https://pbs.twimg.com/profile_images/1285102166709919744/XvlWrJny_400x400.jpg`,
}
