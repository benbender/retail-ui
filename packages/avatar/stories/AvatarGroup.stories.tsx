import * as React from 'react'

import { Avatar, AvatarGroup, AvatarProps } from '../src'

const meta = {
  title: 'Retail-UI/AvatarGroup',
  component: AvatarGroup,
}

export default meta

const src = `https://pbs.twimg.com/profile_images/1285102166709919744/XvlWrJny_400x400.jpg`

const Group = (args: AvatarProps) => (
  <AvatarGroup data-testid={meta.title}>
    <Avatar src={src} {...args} />
    <Avatar title="PJ" size="base" {...args} />
    {/* <Avatar icon={UserSolid} {...args} /> */}
  </AvatarGroup>
)

export const Simple = Group.bind({})
// @ts-ignore
Simple.args = {}
