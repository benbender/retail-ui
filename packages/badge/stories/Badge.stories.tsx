import * as React from 'react'

import { Badge, BadgeProps } from '../src'

const meta = {
  title: 'Retail-UI/Badge',
  component: Badge,
}

export default meta

const Template = (args: BadgeProps) => (
  <Badge {...args} data-testid={meta.title} />
)

export const Simple = Template.bind({})
// @ts-ignore
Simple.args = {
  children: 'Badge',
}
