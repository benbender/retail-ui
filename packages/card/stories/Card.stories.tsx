import { ActionButtons } from '@retail-ui/action-buttons'
import * as React from 'react'

import { Card, CardBody, CardFooter, CardHeader, CardProps } from '../src'

const meta = {
  title: 'Retail-UI/Card',
  component: Card,
}

export default meta

const Template = (args: CardProps) => (
  <Card {...args} data-testid={meta.title}>
    <CardBody>
      <p className="text-sm text-gray-600 dark:text-gray-400">
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod.
      </p>
    </CardBody>
  </Card>
)

export const Simple = Template.bind({})
// @ts-ignore
Simple.args = {}

const TemplateFull = (args: CardProps) => (
  <Card {...args} data-testid={meta.title}>
    <CardHeader>Normal Card</CardHeader>
    <CardBody>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod.
      </p>
    </CardBody>
    <CardFooter className="justify-end">
      <ActionButtons />
    </CardFooter>
  </Card>
)

export const Full = TemplateFull.bind({})
// @ts-ignore
Full.args = {}
