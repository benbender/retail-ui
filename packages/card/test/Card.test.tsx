import { render } from '@retail-ui/test-utils'
import * as React from 'react'

import * as Card from '../stories/Card.stories'

describe('Card', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Card.Simple />)
    expect(asFragment()).toMatchSnapshot()
  })
})
