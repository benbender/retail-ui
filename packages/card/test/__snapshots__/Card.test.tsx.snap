// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<DocumentFragment>
  <div
    class="min-w-0 overflow-hidden rounded-lg transition-shadow duration-150 ease-in relative border border-solid text-base hover:shadow focus:shadow bg-white border-gray-200 dark:bg-gray-800"
    data-testid="Retail-UI/Card"
  >
    <div
      class="px-3 py-2 flex-auto max-h-full relative break-words"
    >
      <div
        class=""
      >
        <p
          class="text-sm text-gray-600 dark:text-gray-400"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
